### PR TITLE
Contributing docs and refactor dev packages

### PR DESCRIPTION
**Pull request by KingDarBoja**
- [x] Place dev and test packages inside setup python file.
- [x] Provide contributing docs for pip and conda developers.

### CODE REVIEW

**Comment by KingDarBoja** [comment]
@jkimbo @Cito ready for review

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+```sh
+conda create -n graphql-sc-dev python=3.8
+```
+
+Then activate the environment with `conda activate graphql-sc-dev`.
+
+Proceed to install all dependencies by running:
+
+```console
+pip install -e.[dev]
```
Comment (jkimbo) [inline CONTRIBUTING.md]:
```suggestion
pip install -e ".[dev]"
```

File: tox.ini [hunk 2]
```diff
@@ -1,38 +1,47 @@
...
     PYTHONPATH = {toxinidir}
-deps =
-    .[test]
+install_command = python -m pip install --ignore-installed {opts} {packages}
+deps = -e.[test]
+whitelist_externals =
+    python
 commands =
-    pytest tests {posargs}
+    pip install -U setuptools
```
Comment (jkimbo) [inline tox.ini]: Why is this necessary?

Comment (KingDarBoja) [reply tox.ini]: Sometimes the tox env crash due to `easy-install.pth` not found on conda environment. This is a known issue and some solution steps can be found [here](https://github.com/ContinuumIO/anaconda-issues/issues/542). Not sure if it is due to conda or pip related error but updating setuptools does help.
